lib/pilot/build_manager.rb

```diff
@@ -7,9 +7,11 @@ def upload(options)
 
       Helper.log.info "Ready to upload new build to TestFlight (App: #{app.apple_id})...".green
 
-      package_path = PackageBuilder.new.generate(apple_id: app.apple_id,
-                                                 ipa_path: config[:ipa],
-                                             package_path: "/tmp")
+      plist = FastlaneCore::IpaFileAnalyser.fetch_info_plist_file(config[:ipa])
+      package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(app_id: app.apple_id,
+                                                                      ipa_path: config[:ipa],
+                                                                  package_path: "/tmp",
+                                                                      platform: plist["DTPlatformName"])
```

Comment:
Fallback already exists in the method being invoked: https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/ipa_upload_package_builder.rb#L24